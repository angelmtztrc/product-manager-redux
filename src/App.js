import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

// components
import Header from './components/Header';
import Products from './components/Products';
import AddProduct from './components/AddProduct';
import EditProduct from './components/EditProduct';

function App() {
  return (
    <Router>
      <Header />
      <main className="container mt-16 mx-auto">
        <Switch>
          <Route exact path="/" component={Products} />
          <Route exact path="/products/new" component={AddProduct} />
          <Route exact path="/products/edit/:id" component={EditProduct} />
        </Switch>
      </main>
    </Router>
  );
}

export default App;
