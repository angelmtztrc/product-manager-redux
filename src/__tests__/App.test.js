import '@testing-library/jest-dom';
import { mount } from 'enzyme';

// components
import App from '../App';

describe('Tests in <App/>', () => {
  let wrapper = mount(<App />);
  test('should render the component successfully', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should render a table with products', () => {
    expect(wrapper.find('Products').exists()).toBe(true);
  });
});
