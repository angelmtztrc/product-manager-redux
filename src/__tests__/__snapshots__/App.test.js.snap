// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <App/> should render the component successfully 1`] = `
Array [
  <nav
    className="py-8 w-full bg-primary"
  >
    <div
      className="items-ce container flex items-center justify-between mx-auto text-white"
    >
      <h1
        className="text-4xl font-bold"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Product Manager - Redux
        </a>
      </h1>
      <a
        className="px-6 py-3 font-bold bg-secondary rounded-lg uppercase"
        href="/products/new"
        onClick={[Function]}
      >
        Add Product
      </a>
    </div>
  </nav>,
  <main
    className="container relative mt-16 mx-auto"
  >
    Array [
      <h2
        className="text-center text-3xl font-semibold"
      >
        Product Listing
      </h2>,
      <div
        className="mt-8"
      >
        <div
          className="grid gap-4 grid-cols-1 animate-pulse"
        >
          <div
            className="col-span-1 px-4 py-2 w-full h-12 bg-gray-300 rounded-lg"
          />
          <div
            className="grid gap-4 col-span-1 grid-cols-3"
          >
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
            <div
              className="col-span-1 px-4 py-2 h-12 bg-gray-300 rounded-lg"
            />
          </div>
        </div>
      </div>,
    ]
  </main>,
]
`;
